integration/fuse/src/main/java/alluxio/fuse/AlluxioJniFuseFileSystem.java

```diff
@@ -395,17 +396,22 @@ private int openInternal(String path, FuseFileInfo fi) {
         return -ErrorCodes.EOPNOTSUPP();
       }
     }
-
+    long fd = mNextOpenFileId.getAndIncrement();
     fi.fh.set(fd);
     try {
-      if (openAction == OpenAction.WRITE_ONLY) {
-        if (status != null) {
-          OpenFlags openFlags = OpenFlags.valueOf(flags);
-          if (openFlags == OpenFlags.O_CREAT || openFlags == OpenFlags.O_EXCL) {
-            return -ErrorCodes.EEXIST();
-          }
-          mFileSystem.delete(uri);
+      // if this is an write operation, and file already exist
+      // we according to O_TRUNC flag to decide whether remove old file
+      // if not specify O_TRUNC, because alluxio can't overwrite part of content of a file, return error for user
+      // attention: according to posix standard, we can't remove file only specify O_WRONLY or O_RDWR flag
+      if (openAction != OpenAction.READ_ONLY && status != null) {
```

Comment:
Can we error out when 
READ_ONLY & truncate are provided together?
e.g.

    if (readOnly && truncate) {
      LOG.error(String.format("Cannot open file %s with flag 0x%x for read and truncate", 
          path, flags);
      return -ErrorCodes.EACCES();
    }
  if (truncate && file exist) {
    deleteFile
  }